### PR TITLE
Document ways to get changed courses and providers

### PR DESCRIPTION
**Pull request by benilovj**

### CODE REVIEW

File: source/index.html.md.erb [hunk 1]
```diff
@@ -184,6 +184,34 @@ Parameter | Description
 --------- | -----------
 recruitment_cycle | 4-character year (e.g. 2019 for 2019/20 courses)
 
+### Get all changed courses
+
+This endpoint retrieves all courses changed since midnight UTC of the specified day.
```
Comment (misaka) [inline source/index.html.md.erb]: ~I don't see how we specify a day. Do you mean the current day the request is made?~

I see it now.

File: source/index.html.md.erb [hunk 2]
```diff
@@ -184,6 +184,34 @@ Parameter | Description
...
+#### HTTP Request
+
+`GET <%= config[:base_api_url] %>/<recruitment_cycle>/courses?changed_since=<YYYYMMDD>`
+
+#### URL Parameters
+
+Parameter | Description
+--------- | -----------
+recruitment_cycle | 4-character year (e.g. 2019 for 2019/20 courses)
+changed_since | Day in YYYYMMDD format
```
Comment (tvararu) [inline source/index.html.md.erb]: ISO8601 is `2019-01-11` even though `20190111` is also allowed ([says Wikipedia?](https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates)), [should we support that instead?](https://xkcd.com/1179/) I think it's slightly more readable.

**Review by tvararu — APPROVED** [review]
Can't spot anything wrong with this

File: source/index.html.md.erb (hunk 2 above)
Comment (misaka) [inline source/index.html.md.erb]: Is there a reason we want to provide an interface to get changes with days-granularity? Do we not need/want to support higher-granularity updates? Every hour or N hours, for example?

I like ZenDesk's implementation for chat messages, and we're going to have to do something like that for paging anyway, so why not just do epoch seconds.

Comment (tvararu) [reply source/index.html.md.erb]: My 2 cents is the thinking is:

- We're not going to have lots of updates happening, a changeset for a given slow day could be only a few dozen records potentially
- Splitting by days makes it easier to work with for mental arithmetic purposes (thinking in seconds isn't super easy)
- You can cache per day at the database level (though we probably won't need to)

Jake wrote down some more context from a chat we had today though so he might be able to chime in with better reasons.

File: source/index.html.md.erb [hunk 3]
```diff
@@ -192,6 +192,34 @@ Parameter | Description
 --------- | -----------
 recruitment_cycle | 4-character year (e.g. 2019 for 2019/20 courses)
 
+### Get all changed courses
+
+This endpoint retrieves all courses changed since midnight UTC of the specified day.
```
Comment (misaka) [inline source/index.html.md.erb]: Looks like this needs to be updated to reflect the switch to seconds. I can update it.

Comment (benilovj) [reply source/index.html.md.erb]: Fixed

File: source/index.html.md.erb (hunk 2 above)
Comment (misaka) [reply source/index.html.md.erb]: @tvararu discussed this some more with @benilovj and we've flipped back to the seconds-resolution thing. I appreciate that this is a lot of flipping back and forth, let's discuss this together. I've updated the documentation with my latest proposal.

**Comment by misaka** [comment]
Having updated the docs just now, I can't help but wonder if there's a reason to include the current paging mechanism (page=1, etc) at all, or whether it shouldn't be completely replaced by the mechanism described in the `Retrieving Changed Records` section I've just added. Would like to have a final discussion about all this if/when we have time today (Tuesday).

**Review by benilovj — COMMENTED** [review]
I'm happy with your proposal about the incremental load, @misaka 👍 

Regarding the all records pagination, shall we look at that on another PR to keep things moving?

File: source/index.html.md.erb [hunk 4]
```diff
@@ -33,24 +33,123 @@ curl "api_endpoint_here"
...
+|               |                                                                     |                                                                                                                                       |
+
 
-Header | Description | Example
------- | ----------- | -------
-Link   | Links to the next and last pages | `Link: <<%= config[:base_api_url] %>/courses?page=169>; rel="last", <<%= config[:base_api_url] %>/courses?page=2>; rel="next"`
-Per-Page | The number of results in the page | `Per-Page: 100`
-Total | The total number of results | `Total: 16888`
+The response headers will include information related to continuing to retrieve
+changed records either immediately, or when ready to injest more records.
```
Comment (tvararu) [inline source/index.html.md.erb]:
```suggestion
changed records either immediately, or when ready to ingest more records.
```

**Review by tvararu — APPROVED** [review]
Looks good. 👍

**Comment by misaka** [comment]
While making these recent changes, I started wondering if slate really is the right thing to use here. Part of it's core feature-set seems to be to accomodate programming-language APIs where we'd provide usage examples in different languages, which we don't do. To do this it constrains us a little by shoving pre-formatted code blocks to the right of the text instead of allowing them to be inline. We could also get away with a plain MarkDown file that's available via our repository.

Anyone else have similar thoughts?

**Pull request merged**